metadoc-js/src/main/scala/metadoc/MetadocTextModelService.scala

```diff
@@ -0,0 +1,55 @@
+package metadoc
+
+import scala.collection.mutable
+import scala.concurrent.ExecutionContext.Implicits.global
+import scala.concurrent.Future
+import scala.meta.internal.semantic.{schema => s}
+import scala.scalajs.js.annotation.ScalaJSDefined
+import monaco.Promise
+import monaco.Uri
+import monaco.common.IReference
+import monaco.editor.Editor
+import monaco.editor.IModel
+import monaco.services.ITextEditorModel
+import monaco.services.ITextModelResolverService
+
+@ScalaJSDefined
+object MetadocTextModelService extends ITextModelResolverService {
+  // NOTE: we have a private cache here to avoid "duplicate model" errors.
+  // It's possible to hit those errors for example in the reference provider, where
+  // we call `Future.sequence(Seq.map(modelReference))`. Since js is single threaded,
+  // we are safe from race conditions.
+  private val cache = mutable.Map.empty[String, IModel]
```

Comment:
`editor.getModel(s)` worked fine until I called `Future.sequence(Seq[Future[IModel]])`, the guard on line 39 is not sufficient since it's possible another callback creates a new model after the guard but before we finish fetching attributes on line 44. Maybe there's a cleaner way to accomplish this, this was a quick hack.